server/player/player.go

```diff
@@ -565,6 +575,10 @@ func (p *Player) Hurt(dmg float64, source damage.Source) (float64, bool) {
 
 		var damageToAttacker int
 		damageToArmour := int(math.Max(math.Floor(dmg/4), 1))
+		// Campfires don't deal armour damage.
+		if s, ok := source.(damage.SourceFire); ok && s.Campfire {
```

Comment:
Why not just return false in `ReducedByArmour` if `s.Campfire` is true?

Reply:
Because the armor still protects you, but it doesn't take any damage. Also, The server still needs to play a Burnign Sound and do everything at the end of the method.